src/instructlab/utils.py

```diff
@@ -564,3 +595,53 @@ def split_hostport(hostport: str) -> tuple[str, int]:
     if not hostname or not port:
         raise ValueError(f"Invalid host-port string: '{hostport}'")
     return hostname, port
+
+
+def convert_messages_to_legacy_dataset(
+    dataset: List[MessageSample],
+) -> List[LegacyMessageSample]:
+    """
+    Converts the new HuggingFace messages dataset format to the legacy format.
+    **Note**: The legacy dataset format assumes only a turn of 1. All extra turns will be dropped.
+    """
+    converted_dataset: List[LegacyMessageSample] = []
+    for dp in dataset:
+        # in new dataset, the roles of a message will be determined.
+        if len(dp["messages"]) < 3:
```

Comment:
```suggestion
        if len(dp["messages"]) != 3:
```

We always expect 3 entries, no less no more, right?

Reply:
I originally thought this, but if in the future we start having multi-turn samples with over 3 messages, we shouldn't immediately discard all of them, but should instead drop everything after the first 3.